internal/util/debug/debug.go

```diff
@@ -122,3 +144,123 @@ func RunHandler(ctx context.Context, addr string, r prometheus.Registerer, l *za
 	s.Close()
 	l.Sugar().Info("Debug server stopped.")
 }
+
+func (obj metric) getValue() float64 {
+	x := prometheusGather()
+	if obj.metricLabel == "" {
+		y := metricRetriever(x, obj.metricName)
+		switch obj.metricType {
+		case 0:
```

Comment:
https://pkg.go.dev/github.com/prometheus/client_model@v0.5.0/go#MetricType

Reply:
since MetricType is of
type MetricType **int32** 

metricType field in the struct gets assigned the typecasted metricFamily.Type as `int` which helps in referencing at lines 152 and later

```
func generateGraphAll(........

finalMetricObj.metricType = int(*metricFamily.Type)
...
}
```